bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/general/GeneralRule.java

```diff
@@ -103,61 +102,49 @@ public IP2Problems check(ObservationElements elements)  {
 
         private static final String WFS_EMPTY_NAME_TEMPLATE = "%s name field should not be empty";
         private static final String P2_PREFERRED = "PWFS2 is the preferred peripheral wavefront sensor.";
-        private static final String NO_AO_OTHER = "Altair cannot use both %s and %s.";
+        private static final String NO_AO_OTHER = "Altair cannot use %s and %s.";
         private static final String NGS_AOWFS = "Altair NGS must have an AOWFS target.";
         private static final String LGS_WFS = "Altair %s must have %s tip-tilt star.";
-
         private static final String LGS_OI_NO_GMOS = "Altair LGS + OI mode requires GMOS-N.";
-
-        // SCT-346: changes the checking and message below to make it more general
-        // Now applies to any WFS
-//        private static final String AOWFS_MESSAGE = "Please make sure that the AOWFS coordinates, proper motions, and " +
-//                "tracking details are exactly the same as those of the target";
         private static final String COORD_CLASH_MESSAGE =
             "Objects with the same name must have the same coordinates.";
         private static final String NAME_CLASH_MESSAGE =
                 "Objects with the same coordinates must have the same name.";
         private static final String TAG_CLASH_MESSAGE =
                 "Objects with the same name must have the same type and coordinates.";
 
-        private static final String WFS_DEFINED_MESSAGE = "WFS should be defined";
-        //TODO: Rule to check if the WFS is in the patrol field, whatever that means
-        //private static final String WFS_OUT_PATROL_FIELD_TEMPLATE = "%s outside of the patrol field";
-
-        private String formatGuiderString(final Iterator<GuideProbe> iter) {
-            if (iter == null || !iter.hasNext())
-                return "";
-
-            final StringBuilder buf = new StringBuilder();
-            buf.append(iter.next().getKey());
-            while (iter.hasNext()) {
-                buf.append(", ").append(iter.next().getKey());
-            }
-            return buf.toString();
+        private String formatGuiderString(final Collection<GuideProbe> probes) {
+            final String startDelim = probes.size() >= 2 ? "[" : "";
+            final String endDelim   = probes.size() >= 2 ? "]" : "";
+            final String sep        = ",";
+            final List<String> probeKeys = probes.stream().map(GuideProbe::getKey).collect(Collectors.toList());
+            return startDelim + String.join(sep, probeKeys) + endDelim;
```

Comment:
I believe I added the previous `formatGuiderString` in my original PR for 2790, but decided to replace it with this one for cleanliness.